engine/src/exec-interface-graphic.cpp

```diff
@@ -713,6 +733,29 @@ void MCGraphic::GetMarkerPoints(MCExecContext& ctxt, uindex_t& r_count, MCPoint*
 
 void MCGraphic::SetMarkerPoints(MCExecContext& ctxt, uindex_t p_count, MCPoint* p_points)
 {
+    if (p_count > 65535)
```

Comment:
In particular, you should only have to call closepolygon once (otherwise you are doing work twice).